src/universal/utils/handleOpenOAuth.ts

```diff
@@ -0,0 +1,45 @@
+import {providerLookup} from '../modules/teamDashboard/components/ProviderRow/ProviderRow'
+import AddProviderMutation from '../mutations/AddProviderMutation'
+
+const handleOpenOAuth = ({
+  name,
+  submitting,
+  submitMutation,
+  atmosphere,
+  onError,
+  onCompleted,
+  teamId
+}) => () => {
+  const {makeUri} = providerLookup[name]
+  const providerState = Math.random()
+    .toString(36)
+    .substring(5)
+  const uri = makeUri(providerState)
+  const popup = window.open(uri)
+  window.addEventListener(
```

Comment:
listen for the code from the popup window. this is necessary to avoid sending meaningful state to the oauth providers. now we just send a random string